app/add_note.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/python2.7
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from google.appengine.api import datastore_errors
+
+from model import *
+from photo import create_photo, PhotoError
+from utils import *
+from detect_spam import SpamDetector
+import extend
+import reveal
+import subscribe
+
+from django.utils.translation import ugettext as _
+from urlparse import urlparse
+
+# TODO(jessien): Clean up duplicate code here and in create.py.
+# https://github.com/google/personfinder/issues/157
+
+# how many days left before we warn about imminent expiration.
+# Make this at least 1.
+EXPIRY_WARNING_THRESHOLD = 7
+
+
+class Handler(BaseHandler):
+
+    def get(self):
+        # Check the request parameters.
+        if not self.params.id:
+            return self.error(404, 'No person id was specified.')
+        try:
+            person = Person.get(self.repo, self.params.id)
+        except ValueError:
+            return self.error(404,
+                _("This person's entry does not exist or has been deleted."))
+        if not person:
+            return self.error(404,
+                _("This person's entry does not exist or has been deleted."))
+        standalone = self.request.get('standalone')
+
+        # Render the page.
+        enable_notes_url = self.get_url('/enable_notes', id=self.params.id)
+
+        self.render('add_note.html',
+                    person=person,
+                    standalone=standalone,
+                    enable_notes_url=enable_notes_url)
+
+
+    def post(self):
+        if not self.params.text:
+            return self.error(
+                200, _('Message is required. Please go back and try again.'))
+
+        if not self.params.author_name:
+            return self.error(
+                200, _('Your name is required in the "About you" section.  '
+                       'Please go back and try again.'))
+
+        if (self.params.status == 'is_note_author' and
+            not self.params.author_made_contact):
+            return self.error(
+                200, _('Please check that you have been in contact with '
+                       'the person after the earthquake, or change the '
```

Comment:
This section was copied from view.py. 
Switched to using "disaster" which is consistent with the person finder homepage descrpition.